SERVER/src/checkpoint_manager.py

```
@@ -1,105 +1,115 @@
+        export_dir = os.path.join(BASE_DATA_PATH, "exports")
+        os.makedirs(export_dir, exist_ok=True)
+
+        if export_format == "json":
+            export_filename = f"story_export_{timestamp}.json"
+            export_path = os.path.join(export_dir, export_filename)
+            story_data = [{"speaker": entry[0], "text": entry[1], "timestamp": entry[2]} for entry in history]
+            try:
+                with open(export_path, "w", encoding="utf-8") as f:
+                    json.dump(story_data, f, indent=4, ensure_ascii=False)
+                return f"Story exported to '{export_filename}' (JSON).", export_filename
+            except Exception as e:
```

Ruff (0.11.9)

Do not catch blind exception: `Exception`

(BLE001)